xc/device.ml

```diff
@@ -1409,29 +1409,46 @@ module Vkbd = struct
 end
 
 module Vusb = struct
-  let call_usb_reset hostbus hostport =
-    let usb_reset_script = "/opt/xensource/libexec/usb_reset.py" in
+  let exec_usb_reset_script argv =
+    let stdout, stderr = Forkhelpers.execute_command_get_output !Xc_resources.usb_reset_script argv in
+    debug "usb_reset_script %s returned stdout=%s stderr=%s" (String.concat " " argv) stdout stderr
+
+  let cleanup domid =
+    try
+      let cmd = ["cleanup"; "-d"; string_of_int domid] in
+      exec_usb_reset_script cmd
+    with _ ->
+      warn "Failed to clean up VM %s" (string_of_int domid)
+
+  let call_usb_reset hostbus hostport domid pid action priv =
```

Comment:
Make the parameter into named parameter, these are all strings at least we can use the names to prevent ordering errors